api/app/controllers/spree/api/stock_items_controller.rb

```diff
@@ -54,7 +54,11 @@ def destroy
       private
 
       def load_stock_location
-        render 'spree/api/shared/stock_location_required', status: 422 and return unless params[:stock_location_id]
+        if !params[:stock_location_id]
```

Comment:
@jhawthorn this means that we don't need the `spree/api/shared/stock_location_required` template anymore.  I wonder if we should remove that file, or deprecate it somehow...? (leave a comment in the rabl..?) Thoughts?